heltour/tournament/admin.py

```diff
@@ -45,19 +47,129 @@ class _BaseAdmin(VersionAdmin):
     change_form_template = 'tournament/admin/change_form_with_comments.html'
     history_latest_first = True
 
+    league_id_field = None
+    league_competitor_type = None
+    allow_all_staff = False
+
+    def has_assigned_perm(self, user, perm_type):
+        return 'tournament.%s_%s' % (perm_type, self.opts.model_name) in user.get_all_permissions()
```

Comment:
`has_perm` doesn't work exactly as it should because of the bug workaround in `auth.py` (see the comments in that file)